### PR TITLE
[Jetpack Content Migration Flow] Add site list to Welcome Screen (3)

### PR DESCRIPTION
**Pull request by ovitrif**
Fixes Partially #17368
Depends on: #17388, which depends on #17387
`Ready for merge` after #17388

Adds the sites list to the Welcome Screen of the Jetpack Migration Flow, which is behind a temporary feature flag.

To test:

0. **Prerequisite:** Toggle the following feature flags `on` in `Wordpress/build.gradle` code:
   (it's not possible to do this from the app because we need to be logged in to switch feature flags):
   - Set `JETPACK_SHARED_LOGIN` to `true`
   - Set `JETPACK_PROVIDER_SYNC` to `true`
   - Set `JETPACK_MIGRATION_FLOW` to `true`
   - Press the `Sync Now` button that appears at the top right of the file editing view in Android Studio
1. Build & Install the WordPress app then Login
   _Note: If testing this right after [dependency PR (2)](https://github.com/wordpress-mobile/WordPress-Android/pull/17388) it's enough to uninstall the Jetpack app and skip step 1._
2. Build & Install the Jetpack app **using the same variant**
   (e.g. if WordPress was `wordpressWasabiDebug` , use `jetpackWasabiDebug`)
3. **Expect** to see the "Welcome to Jetpack!" screen
4. **Verify** that the UI matches the design
   ℹ️ Note: The blur effect behind the buttons is [only available on Android 12 or higher](https://github.com/wordpress-mobile/WordPress-Android/pull/17389#pullrequestreview-1167090705).
   _Note: There's been some recent design updates which are not implemented in this PR._
   1. Scroll down
   2. **Expect** the header to scroll together with the site list.
5. **Expect** Closing and reopening Jetpack app to **not** show the "Welcome to Jetpack" screen
6. **Expect** Logging out and closing Jetpack app app to **not** show the "Welcome to Jetpack" screen

### Preview

<details>
<summary> 🎬 Video preview of the scrolling behavior </summary>

https://user-images.githubusercontent.com/4588074/199570238-e4d32f67-7a3e-4fda-a604-be25c22f38d1.mp4
</details>

| Light | Dark |
| - | - |
|  |  |

### Merge Strategy
1. Merge dependency PR into `trunk`: #17388
   ⚠️ Note: That PR also has dependencies.
3. Remove `Not Ready for Merge` flag
4. Merge this PR

## Regression Notes
1. Potential unintended areas of impact
   N/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   N/a

3. What automated tests I added (or what prevented me from doing so)
   None for now - planned for other PRs.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ovitrif** [comment]
I added a couple of improvements today with the most notable one being to handle accessibility (VoiceOver).
I don't plan to add any other changes to this PR so it should be reviewable in the current state.

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -4244,4 +4244,5 @@ translators: %s: Select control option value e.g: "Auto, 25%". -->
     <string name="jp_migration_welcome_site_found_message">We found your site. Continue to transfer all your data and sign in to Jetpack automatically.</string>
     <string name="jp_migration_continue_button">Continue</string>
     <string name="jp_migration_help_button">Need help?</string>
+
```
Comment (antonis) [inline WordPress/src/main/res/values/strings.xml:4247]: Nitpick: We could remove the empty line addition :)

Comment (ovitrif) [reply WordPress/src/main/res/values/strings.xml:4247]: We can do it, although in fact this was just bringing it back after [it was removed](https://github.com/wordpress-mobile/WordPress-Android/pull/17387/files#diff-6bbab92723bb83a1300e86d496d62b9475a6a1e8b5a5feca27721edfbb09943fR4239-R4240) in #17387

I just checked both on `trunk` before the changes for the Welcome Screen were merged and on `release/21.1` and it seems to be conventionally present.

I was thinking it's best to keep this as it was before the changes, wdyt?

Comment (antonis) [reply WordPress/src/main/res/values/strings.xml:4247]: Sounds good Ovi 👍

**Review by antonis — APPROVED** [review]
Thank you for your work on this @ovitrif 🙇 
Everything worked great for me and the code looks good 🎉 

ps. I've also tested on my older Huawei P9 (Android 7) for any performance related issues and it worked [as expected without the transparency](https://user-images.githubusercontent.com/304044/199757390-b6238a74-3216-4b1b-a608-0f9f57c29a48.png).

**Comment by ovitrif** [comment]
> Thank you for your work on this @ovitrif 🙇 Everything worked great for me and the code looks good 🎉

Many thanks for the whole review effort @antonis 🙇 

> ps. I've also tested on my older Huawei P9 (Android 7) for any performance related issues and it worked [as expected without the transparency](https://user-images.githubusercontent.com/304044/199757390-b6238a74-3216-4b1b-a608-0f9f57c29a48.png).

Good point! We'll have to go with this limitation since the blur effect is only available on Android12+

I'll update the PR description so this is kept in mind for future testing 🙇

**Pull request merged**